lnbits/core/tasks.py

```diff
@@ -1,20 +1,103 @@
 from lnbits.tasks import SseListenersDict, register_invoice_listener
 
 from . import db
-from .crud import get_balance_notify, get_wallet
+from .crud import get_balance_notify, get_total_balance, get_wallet
 from .models import Payment
-from .services import websocketUpdater
+from .services import switch_to_voidwallet, websocketUpdater
 
 api_invoice_listeners: Dict[str, asyncio.Queue] = SseListenersDict(
     "api_invoice_listeners"
 )
 
+killswitch: Optional[asyncio.Task] = None
+watchdog: Optional[asyncio.Task] = None
+
+
+async def register_killswitch():
+    """
+    Registers a killswitch which will check lnbits-status repository
+    for a signal from LNbits and will switch to VoidWallet if the killswitch is triggered.
+    """
+    logger.debug("Starting killswitch task")
+    global killswitch
+    killswitch = asyncio.create_task(killswitch_task())
+
+
+async def unregister_killswitch():
+    """
+    Unregisters a killswitch taskl
+    """
+    global killswitch
+    if killswitch:
+        logger.debug("Stopping killswitch task")
+        killswitch.cancel()
+
+
+async def killswitch_task():
+    while True:
+        if settings.lnbits_killswitch:
+            with httpx.Client() as client:
+                try:
+                    r = client.get(settings.lnbits_status_manifest, timeout=4)
+                    r.raise_for_status()
+                    if r.status_code == 200:
+                        ks = r.json().get("killswitch")
+                        if ks and ks == 1:
+                            logger.error(
+                                "Switching to VoidWallet. Killswitch triggered."
+                            )
+                            await switch_to_voidwallet()
+                except (httpx.ConnectError, httpx.RequestError):
+                    logger.error(
+                        f"Cannot fetch lnbits status manifest. {settings.lnbits_status_manifest}"
+                    )
+                    pass
+        await asyncio.sleep(settings.lnbits_killswitch_interval or 60)
+
+
+async def register_watchdog():
+    """
+    Registers a watchdog which will check lnbits balance and nodebalance
+    and will switch to VoidWallet if the watchdog delta is reached.
+    """
+    logger.debug("Starting watchdog task")
+    global watchdog
+    watchdog = asyncio.create_task(watchdog_task())
+
+
+async def unregister_watchdog():
+    """
+    Unregisters a watchdog task
+    """
+    global watchdog
+    if watchdog:
+        logger.debug("Stopping watchdog task")
+        watchdog.cancel()
+
+
+async def watchdog_task():
+    while True:
+        if settings.lnbits_watchdog:
+            WALLET = get_wallet_class()
+            total_balance = await get_total_balance()
+            error_message, node_balance = await WALLET.status()
+            if error_message:
+                logger.error(f"Error in watchdog task: {error_message}")
+            else:
+                delta = node_balance - total_balance
```

Comment:
- duplicates the logic from `admin_api.api_auditor()`
- suggestion: extract logic to `lnbits/core/services.py` and re-use here and in `api_auditor`